### ISSUE TITLE
Create configuration using expression tree

### ISSUE THREAD

**Issue by gsgabrielsouza** [AutoMapper/AutoMapper#3350]
### Source/destination types

```csharp
String -> Class.String  
String -> Class.Class.String 
```

### Mapping configuration

```csharp
// Mapper.Initialize or just the CreateMap snippet 
```

### Expected behavior
Hello guys !

I'm going through a little problem, i'm using automapper and creating maps from a Expression Tree or Reflection  but i'm not obtain success, i will explain the problem.

### Actual behavior

Basically in this code i'm trying pass a expression tree independent to type of property, but this exception occours in 'MapFrom': 
System.ArgumentOutOfRangeException: 'Only member accesses are allowed. dest => Convert(dest.Shipping.Method, Object) '

Exists a workarround for this ?
### Steps to reproduce

```csharp
   var peDestination = Expression.Parameter(destination.GetType(), "dest");
   var peSource = Expression.Parameter(source.GetType(), "src");

    if (propertyMap.Destination.IndexOf('.') >= 0)
    {
         Expression destinationExpression = peDestination;
         foreach (var navigatinoPropertieDestination in propertyMap.Destination.Split('.'))
               destinationExpression = Expression.Property(destinationExpression,navigatinoPropertieDestination);

          var sourceExpression = Expression.Property(peSource, propertyMap.Source);

var sourceMapFromExpression = Expression.Lambda<Func<TSource, object>>(                                
  Expression.Convert(sourceExpression, typeof(object)), new ParameterExpression[] { peSource });

     //// dest => dest.Property
    var destinationMapFromExpression =
                            Expression.Lambda<Func<TDestination, object>>
                                (Expression.Convert(destinationExpression, typeof(object)), new ParameterExpression[] { peDestination });

map.ForPath(destinationMapFromExpression, a => a.MapFrom(sourceMapFromExpression));
}
```

**Comment by lbargaoanu** [comment]
I think this is better suited for StackOverflow.

**Comment by gsgabrielsouza** [comment]
> I think this is better suited for StackOverflow.

Hello @lbargaoanu , I believe that this could be something incremental in the project, I opened the issue precisely for this discussion, because I attached and debugged it directly in the automapper project realizing that the problem is actually Cast, which is done in the 'GetMembers' method. The way I solved the problem was initially, validating if the node is an 'Expression.Convert' and if so, if there was a possibility to ignore it, and so it worked. I would like to know if this is in fact the best solution or if there is a possibility of opening a fork for this solution.

A better way would be to check inside the node for some function call and ignore it.
